### DISCUSSION TITLE
MDX output?

### DISCUSSION THREAD

**Discussion by ilg-ul** [tgreyuk/typedoc-plugin-markdown#710] [Help]
I wonder how difficult would be to extend the plugin to also generate MDX components (based on a configuration option)?

This would allow, for example, adding `class` definitions to the HTML elements, to allow a fine control of rendering in modern environments like Docusaurus.

There was another `docusaurus-plugin-typedoc-api` project, which achieved something similar, but it is no loner maintained. Perhaps we can inherit some of its good features into this project.

**Comment by tgreyuk** [comment]
It would not be difficult per se, but the markdown plugin focus is to produce markdown and deviating by introducing JSX is not really a path Id like to go down at the moment. This kind of thing would probably be better handled by a docusaurus plugin written in react and using dynamic routing such as docusaurus-plugin-typedoc-api (which like you mention the author is no longer actively maintaining).

**Reply by ilg-ul** [reply]
> This kind of thing would probably be better handled by a docusaurus plugin written in react

I guess you are right.. 

Please don't get me wrong, your plugin is very fine, but the use of plain markdown is somehow limited not by your implementation but by markdown itself.

The limitation I felt was, for example, not being able to add custom CSSs to the reference pages, without affecting the rest of the site. (but I confess that the CSSs intricacies are still a challenge for me).

> using dynamic routing such as docusaurus-plugin-typedoc-api

Yes, probably this would be the more advanced solution.

Perhaps the author of `docusaurus-plugin-typedoc-api` would allow us to reuse his code. (I just checked and it is released under MIT).

One possible solution might be to extend your `docusaurus-plugin-typedoc` and, based on a configuration option, to do the processing locally, without invoking the markdown generator.

I'm also considering writing a `docusaurus-plugin-doxygen` to integrate Doxygen pages. If I manage to do this I'll hopefully get more experience and possibly help you with some work on `docusaurus-plugin-typedoc`, if you decide to extend it.

**Comment by tgreyuk** [comment]
> Please don't get me wrong, your plugin is very fine, but the use of plain markdown is somehow limited not by your implementation but by markdown itself.

Absolutely - and it is something I have certainly considered. If there are some specific enhancements (such as custom css wrappers) then i'd certainly be willing to look into it once I have a bit of time.

> I'm also considering writing a docusaurus-plugin-doxygen to integrate Doxygen pages. If I manage to do this I'll hopefully get more experience and possibly help you with some work on docusaurus-plugin-typedoc, if you decide to extend it.

Sounds promising - keep me posted.

**Reply by ilg-ul** [reply]
I asked the maintainer of Docusaurus his opinion on a `docusaurus-plugin-doxygen` inspired by `docusaurus-plugin-typedoc-api` and apparently this is a realistic option, so I'll go for it (https://github.com/facebook/docusaurus/discussions/10637#discussioncomment-11139802).

**Comment by ilg-ul** [comment]
Hi @tgreyuk 

Did you see the announcement in

- https://github.com/xpack/docusaurus-plugin-doxygen/discussions/1

I would greatly appreciate any suggestions, particularly regarding the structure of the pages, the sidebar, etc, but any other feedback will also be highly appreciated. (please use that discussions page).
